contractcourt/channel_arbitrator.go

```diff
@@ -2086,12 +2120,30 @@ func (c *ChannelArbitrator) checkRemoteDanglingActions(
 			continue
 		}
 
+		// Dust htlcs can be canceled back even before the commitment
+		// transaction confirms. Dust htlcs are not enforceable onchain.
+		// If another version of the commit tx would confirm we either
+		// gain or lose those dust amounts but there is no other way
+		// than cancelling the incoming back because we will never learn
+		// the preimage.
+		if htlc.OutputIndex < 0 {
```

Comment:
nice - also think this should be a new commit since it fixes a different bug, maybe a commit where we fix `checkRemoteDanglingActions` and `checkRemoteDiffActions`?

Reply:
will change the commit structure but I don't think this was an error, we did not make a difference between dust and non dust at this point, which was ok in the old implementation.

Reply:
now we put the dust not in the dangling cat.